ide/session.ml

```diff
@@ -145,23 +145,29 @@ let set_buffer_handlers
     buffer#apply_tag Tags.Script.edit_zone
       ~start:(get_start()) ~stop:(get_stop())
     end in
-  let backto_before_error it =
+  let processed_sentence_just_before_error it =
     let rec aux old it =
-      if it#is_start || not(it#has_tag Tags.Script.error_bg) then old
-      else aux it it#backward_char in
+      if it#is_start then None
+      else if it#has_tag Tags.Script.processed then Some old
+      else if it#has_tag Tags.Script.error_bg then aux it it#backward_char 
+      else None in
     aux it it in
   let insert_cb it s = if String.length s = 0 then () else begin
     Minilib.log ("insert_cb " ^ string_of_int it#offset);
     let text_mark = add_mark it in
     let () = update_prev it in
     if it#has_tag Tags.Script.to_process then
       cancel_signal "Altering the script being processed in not implemented"
+    else if it#has_tag Tags.Script.incomplete then
+      cancel_signal "Altering the script being processed in not implemented"
```

Comment:
Is this the right message?

Reply:
yes, `incomplete` mean "incompletely processed"